src/ssort/_main.py

```diff
@@ -123,6 +127,10 @@ def _on_wildcard_import(**kwargs):
                 )
             else:
                 sys.stderr.write(f"Sorting {str(path)!r}\n")
+                if newline != "\n":
+                    # `newline` argument to `write_text` not supported in python
+                    # 3.8 and earlier.
+                    updated = re.sub("\n", newline, updated)
                 path.write_text(updated, encoding=encoding)
```

Comment:
I think we need to be careful here because by default writing to a file in text mode replaces `\n` characters with `os.linesep`. This means we may perform the `\n` for `\r\n` substitution twice, depending on the value of `os.linesep`. This would lead to lines terminated with `\r\r\n`.

I'm not actually 100% certain that would happen here, as Python may recognize the substitution was already performed and avoid redoing the substitution, but it's best we capture something like this in a test case. It may be as simple as monkey patching `os.linesep` to `\r\n`. Although, you'll need to make sure the built in `open` function actually gets the default line separator from `os.linesep` and not somewhere else.

Here is the relevant documentation for the [open](https://docs.python.org/3/library/functions.html#open) function which seems to indicate the above concerns are relevant:

When writing output to the stream, if newline is None, any '\n' characters written are translated to the system default line separator, [os.linesep](https://docs.python.org/3/library/os.html#os.linesep). If newline is '' or '\n', no translation takes place. If newline is any of the other legal values, any '\n' characters written are translated to the given string.

Reply:
Ultimately, I think we'll want to replace the `write_text` with a `write_bytes` and perform the encoding ourselves. That seems like the most straightforward solution.

Reply:
Yup, you're right.  This won't work on windows, but in a slightly different way from before.  Will switch to `write_bytes`.

Reply:
Done